app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
```

Comment:
why does the req contain the UID of the user? I thought you'd use the response form the /me route to get the user's login (unique in panoptes).